boards/sim/sim/sim/configs/cxxtest/Make.defs

```diff
@@ -86,6 +86,18 @@ else
   LDMODULEFLAGS += -T $(TOPDIR)/libs/libc/modlib/gnu-elf.ld
 endif
 
+# NuttX modules are ELF binaries.
+# Non-ELF platforms like macOS need to use a separate ELF toolchain.
+ifeq ($(CONFIG_HOST_MACOS),y)
+  # eg. brew install x86_64-elf-gcc
```

Comment:
Why we don't use x86_64-elf-gcc for OS self too?

Reply:
because the native binary format is Mach-O.

Reply:
> Mach-O
Can  we use --oformat to tell ld generate elf format?
```
--oformat output-format
ld may be configured to support more than one kind of object file. If your
ld is configured this way, you can use the ‘--oformat’ option to specify the
binary format for the output object file. Even when ld is configured to support
alternative object formats, you don’t usually need to specify this, as ld should
be configured to produce as a default output format the most usual format on
each machine. output-format is a text string, the name of a particular format
supported by the BFD libraries. (You can list the available binary formats
with ‘objdump -i’.) The script command OUTPUT_FORMAT can also specify the
output format, but this option overrides it. See Chapter 5 [BFD], page 69.
```

Reply:
macos' ld doesn't a have -oformat option.
i don't know if it's possible to use gnu ld for mach-o.
it's simpler to use a separate toolchain for different targets.